toolbox

```diff
@@ -30,49 +40,58 @@ toolboxrc="${HOME}"/.toolboxrc
 if [ -f "/etc/default/toolbox" ]; then
-	source "/etc/default/toolbox"
+    source "/etc/default/toolbox"
 fi
 
 # User overrides
 if [ -f "${toolboxrc}" ]; then
-	source "${toolboxrc}"
+    source "${toolboxrc}"
 fi
 
 if [[ -n "${TOOLBOX_DOCKER_IMAGE}" ]] && [[ -n "${TOOLBOX_DOCKER_TAG}" ]]; then
-	TOOLBOX_NAME=${TOOLBOX_DOCKER_IMAGE}-${TOOLBOX_DOCKER_TAG}
+    TOOLBOX_NAME=${TOOLBOX_DOCKER_IMAGE}-${TOOLBOX_DOCKER_TAG}
 else
-	echo "Error: Invalid Docker image name: '${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}'" >&2
-	exit 1
+    echo "Error: Invalid Docker image name: '${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}'" >&2
+    exit 1
 fi
 
 machinename=$(echo "${USER}-${TOOLBOX_NAME}" | sed -r 's/[^a-zA-Z0-9_.-]/_/g')
 machinepath="${TOOLBOX_DIRECTORY}/${machinename}"
 osrelease="${machinepath}/etc/os-release"
+
 if [ ! -f "${osrelease}" ] || systemctl is-failed -q "${machinename}" ; then
-	sudo mkdir -p "${machinepath}"
-	sudo chown "${USER}:" "${machinepath}"
+    sudo mkdir -p "${machinepath}"
+    sudo chown "${USER}:" "${machinepath}"
 
-	# Download and extract the image.
-	sudo --preserve-env docker pull "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}"
-	dcid=$(sudo --preserve-env docker create "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}")
-	sudo --preserve-env docker export -o "${machinepath}/${dcid}.tar" ${dcid}
-	sudo --preserve-env docker rm ${dcid}
-	sudo --preserve-env docker rmi "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}" || true
-	sudo tar xvf "${machinepath}/${dcid}.tar" -C "${machinepath}"
-	sudo rm -f "${machinepath}/${dcid}.tar"
+    if ctr version &> /dev/null; then
+        sudo --preserve-env ctr images pull --platform ${PLATFORM} ${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}
+        sudo ctr images mount --rw --platform ${PLATFORM} ${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG} ${machinepath}
+    elif docker stats --no-stream &> /dev/null; then
+        # Download and extract the image.
+        sudo --preserve-env docker pull "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}"
+        dcid=$(sudo --preserve-env docker create "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}")
+        sudo --preserve-env docker export -o "${machinepath}/${dcid}.tar" ${dcid}
+        sudo --preserve-env docker rm ${dcid}
+        sudo --preserve-env docker rmi "${TOOLBOX_DOCKER_IMAGE}:${TOOLBOX_DOCKER_TAG}" || true
+        sudo tar xvf "${machinepath}/${dcid}.tar" -C "${machinepath}"
+        sudo rm -f "${machinepath}/${dcid}.tar"
+    else
+        echo "No supported container runtime found." >&2
+        exit 2
+    fi
 
-	sudo touch "${osrelease}"
+    sudo touch "${osrelease}"
 fi
 
 # Special case for when SSH tries to pass a shell command with -c
 if [ "x${1-}" == x-c ]; then
-	set /bin/sh "$@"
+    set /bin/sh "$@"
 fi
 
 sudo SYSTEMD_NSPAWN_SHARE_SYSTEM=1 systemd-nspawn \
-	--directory="${machinepath}" \
-	--capability=all \
+    --directory="${machinepath}" \
+    --capability=all \
         ${TOOLBOX_FLAGS} \
```

Comment:
The whitespace diff looks strange at least on GitHub, and I think it would be good to keep the whitespace as it if possible